### PR TITLE
Adding Auto-generate Proxy Configuration

### PR DESCRIPTION
**Pull request by foush**

### CODE REVIEW

File: src/DoctrineMiddleware.php [hunk 1]
```diff
@@ -113,6 +113,7 @@ public function setup()
         if (!$this->setupMetadataDriver($config)) {
             throw new \RuntimeException('No Metadata Driver defined');
         }
+        $config->setAutoGenerateProxyClasses($this->getOption('auto_generate_proxies', true) ? true : false);
```
Comment (juliangut) [inline src/DoctrineMiddleware.php]: Can you simplify to ...?

``` php
$config->setAutoGenerateProxyClasses($this->getOption('auto_generate_proxies', true) === true);
```

Comment (foush) [inline src/DoctrineMiddleware.php]: Can you tell me what benefit we're gaining by doing it that way?

My thought process for `$this->getOption('auto_generate_proxies', true) ? true : false` was margin of error; if somebody were to put 1 or 0 as their `auto_generate_proxies` config value, the ternary would resolve to the correct boolean value where the triple-equals would not.

Comment (juliangut) [inline src/DoctrineMiddleware.php]: When you specify in the documentation:
`auto_generate_proxies` bool indicating whether Doctrine should auto-generate missing proxies (default: true)

I'd really expect the value to be a boolean and not rely in the dynamic nature of PHP, put another way:

``` php
$config = [
    'doctrine' => [
        'connection' => [],
        'auto_generate_proxies': 'I do not want this activated',
    ],
];
```

The string is cast to `true`, that's awkward

Replacing triple-equals to double would accomplish what you want without ternary operator

``` php
$config->setAutoGenerateProxyClasses($this->getOption('auto_generate_proxies', true) == true);
```

Comment (foush) [inline src/DoctrineMiddleware.php]: I see your point. I've updated the PR and replaced the ternary with double-equals.

**Pull request merged**